GitCommands/Git/GitModule.cs

```diff
@@ -2143,42 +2143,62 @@ public IReadOnlyList<PatchFile> GetInteractiveRebasePatchFiles()
 
             List<PatchFile> patchFiles = new();
 
-            if (todoCommits is not null)
+            if (todoCommits is null)
             {
-                string commentChar = EffectiveConfigFile.GetString("core.commentChar", "#");
+                return patchFiles;
+            }
 
-                string? currentCommitShortHash = File.Exists(CurrentFilePath) ? File.ReadAllText(CurrentFilePath).Trim() : null;
-                var isCurrentFound = false;
-                foreach (string todoCommit in todoCommits)
-                {
-                    if (todoCommit.StartsWith(commentChar))
-                    {
-                        continue;
-                    }
+            string commentChar = EffectiveConfigFile.GetString("core.commentChar", "#");
 
-                    string[] parts = todoCommit.Split(Delimiters.Space);
+            // Filter comment lines and keep only lines containing at least 3 columns
+            // (action, commit hash and commit subject -- that could contains spaces and be cut in more --)
+            // ex: pick e0d861716540aa1ac83eaa2790ba5e79988b9489 this is the commit subject
+            string[][] todoCommitsInfos = todoCommits.Where(l => !l.StartsWith(commentChar))
+                .Select(l => l.Split(Delimiters.Space))
+                .Where(p => p.Length >= 3)
+                .ToArray();
 
-                    if (parts.Length < 3)
-                    {
-                        continue;
-                    }
+            string? currentCommitShortHash = File.Exists(CurrentFilePath) ? File.ReadAllText(CurrentFilePath).Trim() : null;
+            bool isCurrentFound = false;
 
-                    string commitHash = parts[1];
-                    CommitData? data = _commitDataManager.GetCommitData(commitHash, out var error);
-                    var isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
-                    isCurrentFound |= isApplying;
+            // retrieve commits bodies to display them in the grid or as tooltips
+            RevisionReader reader = new(this, hasReflogSelector: false, allBodies: true);
+            CancellationTokenSource cts = new(TimeSpan.FromSeconds(15));
```

Comment:
This might be too short. It can take a few seconds for a few commits on a workstation.

Reply:
Is it something you saw?
Because, in GE repo, I made a rebase of 12k commits and it took only 250ms that's why I concluded that in this case 15s was enough. But I have no problem to increase it. 30s?

Reply:
git-log+parsing requires about 2 s for 100K commits so rebasing all commits in the linux repo would require 20s or so on my machine. 30s is safe

The git-rebase operation can require some time though

Reply:
The cancelation token is only for the 'git log & parsing' so I'm happy you confirm that 15s is enough (to rebase 700k commits!) 🤣
But secured 30s is OK for me as we are talking about a worst case scenario that will nearly never happen...

Reply:
> Is it something you saw?

Yes, perhaps caused by hyperactive enterprise AV and / or Windows Update.

(Also git itself became quirky later yesterday. It reproducibly returned exit code 128 without error output when resetting single files using `git checkout` - vanished after reboot.)